examples/termination/main.rs

```diff
@@ -156,6 +172,91 @@ fn rcc_main(rcc_binary_path: Utf8PathBuf) -> AnyhowResult<()> {
+        ],
+        envs_rendered_obfuscated: vec![],
+        retry_strategy: RetryStrategy::Complete,
+    };
+    let conda_environment = Environment::CondaFromManifest(CondaEnvironmentFromManifest {
+        micromamba_binary_path,
+        manifest_path: cargo_manifest_dir.join("examples/termination/conda.yaml"),
+        root_prefix: test_dir.join("micromamba_root"),
+        prefix: test_dir.join("conda_env"),
+        build_timeout: 1200,
+        build_runtime_directory: test_dir.clone(),
+    });
+    let session = Session::Current(CurrentSession {});
+    assert!(matches!(
+        conda_environment
+            .build("unused_id", &session, Utc::now(), &CancellationToken::new())
+            .unwrap(),
+        BuildOutcome::Success(_),
+    ));
+    println!("Finished environment build.");
+    let token = CancellationToken::new();
+    let thread_token = token.clone();
+    let running = thread::spawn(move || {
+        run_attempts_with_rebot(
+            &robot,
+            "test",
+            &conda_environment,
+            &session,
+            20,
+            &thread_token,
+            &test_dir,
+        )
+    });
+    while !flag_file.exists() {
+        // Wait for all children to be created
+        thread::sleep(Duration::from_millis(250));
+        if running.is_finished() {
+            panic!("{:?}", running.join());
+        }
+    }
+    #[cfg(unix)]
+    // 203317 Run "termination" Some("/home/joergherbel/git/robotmk/target/x86_64-unknown-linux-gnu/debug/examples/termination")
+    // -205082 Sleep "micromamba" Some("/home/joergherbel/tmp/micromamba_fun/micromamba")
+    // --205085 Sleep "python" Some("/tmp/.tmpgCIOOH/conda_env/bin/python3.12")
+    // ---205171 Sleep "python" Some("/tmp/.tmpgCIOOH/conda_env/bin/python3.12")
+    assert_eq!(check_tree_size(&mut system, current_pid), 4);
+    #[cfg(windows)]
+    // 9856 Run "termination.exe" Some("\\\\VBoxSvr\\robotmk\\target\\x86_64-pc-windows-msvc\\debug\\examples\\termination.exe")
+    // -6500 Run "micromamba.exe" Some("C:\\Users\\vagrant\\Downloads\\micromamba.exe")
+    // --9396 Run "cmd.exe" Some("C:\\Windows\\System32\\cmd.exe")
+    // ---3236 Run "python.exe" Some("C:\\Users\\vagrant\\AppData\\Local\\Temp\\.tmpVnsIpp\\conda_env\\python.exe")
+    // ----5032 Run "python.exe" Some("C:\\Users\\vagrant\\AppData\\Local\\Temp\\.tmpVnsIpp\\conda_env\\python.exe")
+    assert_eq!(check_tree_size(&mut system, current_pid), 5);
+    assert!(resource_file.exists());
+    token.cancel();
+    match running.join().unwrap() {
+        Err(error) => {
+            let message = format!("{error:?}");
+            assert!(message.starts_with("Cancelled"), "Message: {message}")
+        }
+        ok => panic!("Cancellation failed: {ok:?}"),
+    };
+    assert_eq!(check_tree_size(&mut system, current_pid), 1);
+    // On Windows, we always kill the process tree, so no teardown will run.
     #[cfg(unix)]
     assert!(!resource_file.exists());
     Ok(())
```

Comment:
Can't review it, but please move the removal of executable bits on `examples/termination/process_tree.rs` to a separate PR.

Reply:
Done